master/internal/command/authz_iface.go

```diff
@@ -32,6 +34,27 @@ type NSCAuthZ interface {
 	CanSetNSCsPriority(
 		ctx context.Context, curUser model.User, workspaceID model.AccessScopeID, priority int,
 	) error
+
+	// Tensorboard functions
+	// GET /api/v1/tensorboards/:tb_id
+	CanGetTensorboard(
+		ctx context.Context, curUser *model.User, tb *tensorboardv1.Tensorboard,
+	) (canGetTensorboard bool, serverError error)
+
+	// GET /api/v1/tensorboards
+	FilterTensorboards(
+		ctx context.Context, curUser *model.User, tensorboards []*tensorboardv1.Tensorboard,
+	) ([]*tensorboardv1.Tensorboard, error)
+
+	// POST /api/v1/tensorboards/:tb_id/kill
+	CanTerminateTensorboard(
+		ctx context.Context, curUser *model.User, tb *tensorboardv1.Tensorboard,
+	) error
+
+	// POST /api/v1/tensorboards/:tb_id/set_priority
+	CanSetTensorboardPriority(
+		ctx context.Context, curUser *model.User, tb *tensorboardv1.Tensorboard,
+	) error
```

Comment:
do we want these to take in the whole tensorboardv1.Tensorboard or stay closer to what the NSC interface is asking for? aka workspaceId. Imo these could be more like the existing NSC ones

Reply:
my bad, I misread the code - I thought that it would require another database query, so I decided to keep it this way in order to minimize the queries. I just looked more closely and it appears workspace ID works. Will spend a few more minutes to see what I can combine with the NSC interfaces.